### PR TITLE
services: added communities transfer service

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1677

### CODE REVIEW

File: invenio_rdm_records/services/communities/service.py [hunk 1]
```diff
@@ -355,3 +358,49 @@ def set_default(self, identity, id_, data, uow):
...
+    def _include_one(self, record, community, set_default=False):
+        """This service call differs from the `add` method in that it does not create a request to include the record in the community.
+
+        It adds the record to the community directly.
+        It should only be called from within the service.
+        """
+        default = set_default or not record.parent.communities
+
+        parent_community = getattr(community, "parent", None)
+        already_in_parent = parent_community and str(parent_community.id) in record
```
Comment (jrcastro2) [inline invenio_rdm_records/services/communities/service.py:400]:
```suggestion
        already_in_parent = parent_community and str(parent_community.id) in record.parent.communities
```

File: invenio_rdm_records/services/communities/service.py [hunk 2]
```diff
@@ -355,3 +358,49 @@ def set_default(self, identity, id_, data, uow):
...
+
+        if parent_community and not already_in_parent:
+            record.parent.communities.add(parent_community, request=self.request)
+
+        record.parent.communities.add(community, request=None, default=default)
```
Comment (jrcastro2) [inline invenio_rdm_records/services/communities/service.py:405]: Question: Do we have the use case that the community is already part of the record? (might be worth to add the same check as above)

File: invenio_rdm_records/requests/community_transfer.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from invenio_requests.customizations import RequestType, actions
+
+
+class AcceptCommunityTransfer(actions.AcceptAction):
+    """Represents an accept action used to accept a communtiy transfer request."""
+
+    def execute(self, identity, uow):
+        """Executes approve action."""
+        # Step 1 - move communities to new target using communities service
+        # Step 2 - move records to new parent using community records service
```
Comment (jrcastro2) [inline invenio_rdm_records/requests/community_transfer.py:18]: Major: I guess this is missing?

**Review by jrcastro2 — COMMENTED** [review]
Pair reviewed with @0einstein0

File: invenio_rdm_records/services/communities/service.py [hunk 4]
```diff
@@ -355,3 +358,49 @@ def set_default(self, identity, id_, data, uow):
...
+                ParentRecordCommitOp(
+                    record.parent,
+                    indexer_context=dict(service=current_rdm_records_service),
+                )
+            )
+            uow.register(
+                RecordCommitOp(
+                    record,
+                    indexer=current_rdm_records_service.indexer,
+                    index_refresh=True,
```
[the remark this answers is not in the document]
Comment (alejandromumo) [reply invenio_rdm_records/services/communities/service.py:387]: This code comes from [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/requests/community_inclusion.py#L75)

And yeah, we refresh the index to make sure we get the latest version updated immediately

File: invenio_rdm_records/services/communities/service.py [hunk 5]
```diff
@@ -355,3 +358,49 @@ def set_default(self, identity, id_, data, uow):
...
+                )
+            )
+
+    def _include_one(self, record, community, set_default=False):
+        """This service call differs from the `add` method in that it does not create a request to include the record in the community.
+
+        It adds the record to the community directly.
+        It should only be called from within the service.
+        """
+        default = set_default or not record.parent.communities
```
[the remark this answers is not in the document]
Comment (alejandromumo) [reply invenio_rdm_records/services/communities/service.py:397]: The idea here is to set the community as `default` implicitly when the community is the first one of the record.

File: invenio_rdm_records/services/communities/service.py (hunk 1 above)
Comment (alejandromumo) [reply invenio_rdm_records/services/communities/service.py:400]: Good catch, thanks!

File: invenio_rdm_records/services/communities/service.py (hunk 2 above)
Comment (alejandromumo) [reply invenio_rdm_records/services/communities/service.py:405]: Good one, I will add a check and tests. Thanks!

File: invenio_rdm_records/services/communities/service.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (slint) [reply invenio_rdm_records/services/communities/service.py:387]: On a second look `ParentCommitOp` has [a `bulk_index` param](https://github.com/inveniosoftware/invenio-drafts-resources/blob/116f25479db770312ee9672104d89a587c497cc1/invenio_drafts_resources/services/records/uow.py#L21) for bulk indexing all record and draft versions.

**Pull request merged**